File: backend/models/TrainingHistory.ts
import mongoose from 'mongoose';

export interface ITrainingHistory {
  userId: string;
  username: string;
  collectionName: string;
  documentName: string;
  action: 'upload' | 'delete' | 'create_collection' | 'update_collection' | 'delete_collection' | 'crawl_url';
  timestamp: Date;
  details?: Record<string, any>;
}

const trainingHistorySchema = new mongoose.Schema({
  userId: { type: String, required: true },
  username: { type: String, required: true },
  collectionName: { type: String, required: true },
  documentName: { type: String },
  action: { 
    type: String, 
    enum: ['upload', 'delete', 'create_collection', 'update_collection', 'delete_collection', 'crawl_url'],
    required: true 
  },
  timestamp: { type: Date, default: Date.now },
  details: { 
    type: mongoose.Schema.Types.Mixed,
    // สำหรับ crawl_url จะเก็บข้อมูลเพิ่มเติมเช่น:
    // - modelId: string
    // - pagesProcessed: number
    // - depth: number
    // - success: boolean
    // - error?: string
  }
}, { timestamps: true });

const TrainingHistory = mongoose.model('TrainingHistory', trainingHistorySchema);
export { TrainingHistory }; 

File: backend/models/training_history.py
from pydantic import BaseModel, Field
from typing import Optional, Dict, Any
from datetime import datetime
from enum import Enum

class TrainingAction(str, Enum):
    UPLOAD = 'upload'
    DELETE = 'delete'
    CREATE_COLLECTION = 'create_collection'
    UPDATE_COLLECTION = 'update_collection'
    DELETE_COLLECTION = 'delete_collection'

class TrainingHistory(BaseModel):
    userId: str
    username: str
    collectionName: str
    documentName: Optional[str] = None
    action: TrainingAction
    timestamp: datetime = Field(default_factory=datetime.utcnow)
    details: Optional[Dict[str, Any]] = None

    class Config:
        populate_by_name = True
        json_schema_extra = {
            "example": {
                "userId": "user123",
                "username": "johndoe",
                "collectionName": "my-collection",
                "action": "upload",
                "documentName": "document1.pdf"
            }
        } 